plugins/org.eclipse.papyrus.uml.interaction.model/src-gen/org/eclipse/papyrus/uml/interaction/internal/model/impl/MInteractionImpl.java

```diff
@@ -347,15 +348,19 @@ MMessageImpl addMessage(MMessageImpl message) {
 	 * @generated NOT
 	 */
 	@Override
-	public Optional<? extends MElement<?>> getBottomMostElement() {
+	public Optional<? extends MElement<? extends Element>> getBottommostElement() {
 		Stream<MMessageEnd> messageEnds = getMessages().stream()
 				.flatMap((m) -> Stream.of(m.getSend(), m.getReceive())).filter(Optional::isPresent)
 				.map(Optional::get);
-		return messageEnds.max(Comparator.comparingInt(this::bottomOfMMessageEnd));
+		Stream<MOccurrence<?>> occurrances = getLifelines().stream()
+				.flatMap((lifeline) -> lifeline.getExecutions().stream())
+				.flatMap((e) -> Stream.of(e.getStart(), e.getFinish())).filter(Optional::isPresent)
+				.map(Optional::get);
+		return Stream.concat(messageEnds, occurrances).max(Comparator.comparingInt(this::bottomOfMElement));
```

Comment:
I think that once we add support for other things that cover lifelines (combined fragments, state invariants, interaction references) then the `MLifeline` API will need some union feature that simplifies this.